kernel_netlink.c

```diff
@@ -1086,22 +1086,31 @@ kernel_route(int operation, int table,
         *(int*)RTA_DATA(rta) = ifindex;
 
 #define ADD_IPARG(type, addr) \
```

Comment:
You could refactor the two lines
```c
rta = RTA_NEXT(rta, len); \
rta->rta_type = type; \
```
by putting them inside the `do {  } while` but before the `if` block, and have a nicer flow
```c
if(v4mapped(addr)) { /* … */ } else if(type == RTA_VIA) { /* … */  } else { /* … */  }
```

Reply:
This duplication was already there, but I agree it would look nicer.